### ISSUE TITLE
package garak for kali

### ISSUE THREAD

**Issue by leondz** [NVIDIA/garak#1036]
goal: get a garak `.deb` into kali

* https://www.kali.org/docs/tools/submitting-tools/
* https://bugs.kali.org/view.php?id=8956
* https://www.kali.org/docs/development/intermediate-packaging-example/#dependenciesmaintainers

spec:

* targeting pythons 3.10, 3.11, 3.12
* targeting arch  `x86_64`, `arm64`

scope:

* work out how to package/shrink garak
* generate a kali-compatible deb
* submit and guide the package's first release

out of scope / future work:

* work out github workflows to build a new package and ping kali so that doing releases automatically updates what's in kali

**Comment by jmartin-tech** [comment]
Check of requirements based on 0b837c1 and current `kali-rolling` dependencies per [advanced packaging docs](https://www.kali.org/docs/development/advanced-packaging-example/), finds a number of packages that may need additional support:
```
--MISSING base2048--
--MISSING transformers--
--MISSING datasets--
python3-colorama - Cross-platform colored terminal text in Python - Python 3.x
python3-tqdm - fast, extensible progress bar for Python 3 and CLI tool
--MISSING cohere--
python3-openai - OpenAI Python API library
--MISSING replicate--
--MISSING google-api-python-client--
python3-backoff - Function decoration for backoff and retry
python3-rapidfuzz - Rapid fuzzy string matching library (Python library)
python3-jinja2 - small but fast and easy to use stand-alone template engine
python3-nltk - Python3 libraries for natural language processing
--MISSING accelerate--
--MISSING avidtools--
--MISSING stdlibs--
--MISSING langchain--
--MISSING nemollm--
--MISSING octoai-sdk--
python3-cmd2 - Enhanced Python cmd module - Python 3.x
--MISSING torch--
python3-sentencepiece - SentencePiece binding for Python3
python3-markdown - text-to-HTML conversion library/tool (Python 3 version)
python3-numpy - Fast array facility to the Python language (Python 3)
--MISSING zalgolib--
--MISSING ecoji--
--MISSING deepl--
--MISSING fschat--
--MISSING litellm--
python3-jsonpath-ng - Robust and extended implementation of JSONPath for Python
--MISSING huggingface_hub--
--MISSING python-magic-bin--
--MISSING python-magic--
--MISSING lorem--
--MISSING xdg-base-dirs--
--MISSING wn--
--MISSING ollama--
python3-tiktoken - fast BPE tokeniser for use with OpenAI's models
```

There are 25 packages that are not currently available pre-wrapped for Kali dependency usage, this can be worked around with some sort of mixed `venv` embedded in the package however this may require significant maintenance over time.

**Comment by leondz** [comment]
Sounds like it might make sense to only work out one single release for
now. Would like to include `ansiescape` in that if possible.

**Comment by jmartin-tech** [comment]
Deferring from original milestone to find a more approachable packaging strategy that may lead to a lower maintenance burden and possibly improve on the installed base storage requirements. The ultimate goal would be for this project to be considered for inclusion as a default installed package in an Offensive Security targeted OS.

**Comment by leondz** [comment]
smaller footprint from deferred loading changes may help here. may be worth revisiting this after the summer. LLM work tends to bring in "exotic" packages so the contemporary immaturity of the field may be a cause of friction here for a little while yet.

**Comment by leondz** [comment]
revisit after #1199 lands
